### ISSUE TITLE
Mispelled words highlight is hard to see because done via *highlight* instead of <inverted color>highlight<inverted color> 

### ISSUE THREAD

**Issue by idvorkin** [GNUAspell/aspell#590]
*TL;DR - ncurses isn't being linked in the aspell brew formula.  Investigating*

I spell check files with very long lines  (wraps to 5 vertical lines) and several bad words. On my windows machines (WSL via Windows Terminal ) the entire line is printed with the bad word highlighted via ``` *misspelled*```. This makes it really hard to see the misspelling. 

Using ispell, only a small set of words is shown for context (e.g. 10 before and after) and the misspelled word context is displayed using inverted colors, which makes it easy to see. 

On one of my machines (ish on iPad), I get inverted colors for misspelled words, so I think there's something terminal related, but not sure how to debug/fix.  

Is there a way to: 

1) Debug the problem. 
2) Option to set highlight mode
3) Option for how many words to add with context
4) Better question to be asking?

In case it helps
```
➜  750words git:(master) ✗ set | grep TERM=
TERM=xterm
ZSH_TMUX_FIXTERM=true
ZSH_TMUX_TERM=screen
➜  750words git:(master) ✗ aspell --version
@(#) International Ispell Version 3.1.20 (but really Aspell 0.60.8)
➜  750words git:(master) ✗

```

**Comment by kevina** [comment]
It's likely proper curses support was not compiled into the Windows version you are using.  Have a look at the Win32 notes (http://aspell.net/man-html/WIN32-Notes.html) and in particular the section on (PD)Curses.

**Comment by idvorkin** [comment]
Curses is a good hypothesis.   This is on WSL (Windows Subsystem for Linux)
using ubuntu with aspell installed via brew, so I'd imagine it's either
incorrectly built/linked within the brew package or there is something else
going on. Can you recommend a way to verify if my binary is using the right
curses library (or other things to check)?

**Comment by kevina** [comment]
I am not familiar with WSL but in Linux you can use:
```
ldd `which aspell`
```
To check which libraries the aspell binary is Linux linked with, you should see something like
```
	linux-gate.so.1 =>  (0xb77ae000)
	libaspell.so.15 => /usr/lib/libaspell.so.15 (0xb76d6000)
	libncursesw.so.5 => /lib/i386-linux-gnu/libncursesw.so.5 (0xb76a6000)
	libtinfo.so.5 => /lib/i386-linux-gnu/libtinfo.so.5 (0xb7686000)
	libstdc++.so.6 => /usr/lib/i386-linux-gnu/libstdc++.so.6 (0xb75a1000)
	libc.so.6 => /lib/i386-linux-gnu/libc.so.6 (0xb73f7000)
	libdl.so.2 => /lib/i386-linux-gnu/libdl.so.2 (0xb73f2000)
	libm.so.6 => /lib/i386-linux-gnu/libm.so.6 (0xb73c6000)
	/lib/ld-linux.so.2 (0xb77af000)
	libgcc_s.so.1 => /lib/i386-linux-gnu/libgcc_s.so.1 (0xb73a7000)
```
Here my aspell binary is linked with `libncursesw.so.5`.  If nothing with curses in the name is listed then it is likely Aspell was not compiled with curses support.

**Comment by idvorkin** [comment]
You're right, ncurses isn't linked in. Here's what I get: 
```
 idvor ldd `which aspell`
        linux-vdso.so.1 (0x00007fffc8f54000)
        libaspell.so.15 => /home/linuxbrew/.linuxbrew/Cellar/aspell/0.60.8/lib/libaspell.so.15 (0x00007ffc0bb30000)
        libstdc++.so.6 => /home/linuxbrew/.linuxbrew/lib/libstdc++.so.6 (0x00007ffc0b760000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007ffc0b360000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007ffc0b150000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007ffc0adb0000)
        /home/linuxbrew/.linuxbrew/lib/ld.so => /lib64/ld-linux-x86-64.so.2 (0x00007ffc0c000000)
        libgcc_s.so.1 => /home/linuxbrew/.linuxbrew/Cellar/gcc/5.5.0_4/lib/libgcc_s.so.1 (0x00007ffc0ab80000)
'''

I'm new to brew debugging, here's the package definition, I wonder if it's missing some flags: 
https://github.com/Homebrew/homebrew-core/blob/master/Formula/aspell.rb

**Comment by idvorkin** [comment]
When I built from the source referenced in the brew formula, it links ncurses, and highlights work as expected.

**Comment by idvorkin** [comment]
Hey @SMillerDev I saw you last updated the brew formula for aspell, any ideas?

**Comment by SMillerDev** [comment]
Homebrew will not pull in homebrew dependencies unless they're explicitly defined. You could make a pull request to homebrew-core to add `uses_from_macos "ncurses"` to the aspell formula. That way it'll depend on it on Linux and use the system one on macOS.

**Comment by idvorkin** [comment]
Thanks. I'm making the PR now, how should I think about when to add a regular dependency vs using uses_from_macos? Also, I don't have OSX, how do folks normally test multi-platform?

**Comment by kevina** [comment]
You may continue the conversation but since this is not a bug in Aspell itself I am closing this.

**Comment by SMillerDev** [comment]
Uses_from_macos doesn't do anything on macOS so no need to test it there. But CI will check your PR regardless.

**Comment by idvorkin** [comment]
Fixing via https://github.com/Homebrew/homebrew-core/pull/48163
